omegaconf/omegaconf.py

```diff
@@ -411,12 +412,14 @@ def register_new_resolver(
         name: str,
         resolver: Resolver,
         *,
-        use_cache: Optional[bool] = False,
+        use_cache: bool = False,
     ) -> None:
         """
         Register a resolver.
 
-        :param name: Name of the resolver.
+        :param name: Name of the resolver. If this name has already been registered,
+            then a `ValueError` is raised (unless `resolver` is the same callable object
+            and `use_cache` has the same value as previously registered).
```

Comment:
I think we should simplify here and not add the nice shortcut.
We will probably make caching more complicated in the future, which will also make this handling more complicated (by adding additional keyword to register_resolver).

Le'ts just raise an exception if a resolver with the same name is used even if it's registering the same callable.

Reply:
Done by reverting previous changes, new version is in 3554985